src/ray/object_manager/object_store_notification_manager.h

```diff
@@ -28,8 +28,11 @@ class ObjectStoreNotificationManager {
   ///
   /// \param io_service The asio service to be used.
   /// \param store_socket_name The store socket to connect to.
+  /// \param exit_on_error The manager will exit with error when it fails
+  ///                      to process messages from socket.
   ObjectStoreNotificationManager(boost::asio::io_service &io_service,
-                                 const std::string &store_socket_name);
+                                 const std::string &store_socket_name,
+                                 bool exit_on_error = true);
```

Comment:
Not a strong preference, but I think it would be good to make this an explicit argument instead of optional.

Reply:
yeah this is a special case I want to keep backward compatibility.